compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/lower-context-selector-simple.expect.md

```diff
@@ -0,0 +1,38 @@
+
+## Input
+
+```javascript
+// @enableLowerContextAccess
+function App() {
+  const {foo, bar} = useContext(MyContext);
+  return <Bar foo={foo} bar={bar} />;
+}
+
+```
+
+## Code
+
+```javascript
+import { c as _c } from "react/compiler-runtime"; // @enableLowerContextAccess
+function App() {
+  const $ = _c(3);
+  const { foo, bar } = useContext(MyContext, _temp);
+  let t0;
+  if ($[0] !== foo || $[1] !== bar) {
+    t0 = <Bar foo={foo} bar={bar} />;
+    $[0] = foo;
+    $[1] = bar;
+    $[2] = t0;
+  } else {
+    t0 = $[2];
+  }
+  return t0;
+}
+function _temp(t0) {
+  return [t0.foo, t0.bar];
+}
```

Comment:
How are you thinking about sequencing the next steps here? Asking because the generated code doesn't seem like it could work yet: the temp function returns an array, but the context value is destructured into an object and there's nothing that tells us how to map the array into the object.

Seems like we'd need to convert the destructuring into an ArrayPattern to match? Also, if we take Jack's suggestion of compiling into a custom `useContextSelector`-style hook, then we can implement a basic version of that in shared-runtime and this could be an end-to-end test with sprout.

Reply:
The return value of the selector isn't returned from the useContext call. The runtime simply compares the return values from the selector to determine if the context is dirty. If the values are different, then it returns the _context object_ from useContext call, not the values from the selector function. 

Ideally we'd just compile the selector function with Forget and compare the result, rather than iterative over an array, but that's optimisation for the future.

Reply:
ahhhhh got it, makes sense